tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php

```diff
@@ -529,4 +529,39 @@ private function get_connected_user() {
 
 		return $user;
 	}
+
+	public function test_ge111t_remote_authorize_url() {
+		// Arrange
+		$_GET['utm_source'] = 'test-source';
+		$_GET['utm_medium'] = 'test-medium';
+		$_GET['utm_campaign'] = 'test-campaign';
+		$_GET['utm_not_allowed_param'] = 'test-test';
+
+		$utm_campaign = [
+			'source' => 'transient-source',
+			'medium' => 'transient-medium',
+			'campaign' => 'transient-campaign',
+		];
+
+		set_transient( 'elementor_core_campaign', $utm_campaign );
+
+		// Act
+		$url = $this->app_stub->proxy_get_remote_authorize_url();
+
+		// Assert
+		$parsed_url = parse_url( $url );
+		$parsed_query_params = [];
+		parse_str( $parsed_url['query'], $parsed_query_params );
```

Comment:
Maybe extract to a method?

Reply:
What exactly do you mean? Two variables have been created.